### DISCUSSION TITLE
>TCPChannelResourceBasic::send    problem

### DISCUSSION THREAD

**Discussion by anyingbinglong** [eProsima/Fast-DDS#3596] [Q&A]
> 

I have a problem:  asio::write()   -----   why  can not   use asio::async_write ？

_Originally posted by @anyingbinglong in https://github.com/eProsima/Fast-DDS/issues/947#issuecomment-1592544398_

**Comment by JesusPoderoso** [comment]
Hi @anyingbinglong 

This method,  as part of the internal implementation, is part of internal desing.

I am moving this ticket to the Support discussion forum as per [Fast DDS Contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support). This is not a bug but a question.
